src/Modules/Visualization/ShowFieldGlyphs.cc

```diff
@@ -167,29 +177,41 @@ void GlyphBuilder::addGlyph(
       glyphs.addNeedle(p1, p2, node_color, node_color);
       break;
     case RenderState::GlyphType::COMET_GLYPH:
-      glyphs.addSphere(p2, radius, resolution, node_color);
-      glyphs.addCone(p2, p1, radius, resolution, false, node_color, node_color);
+    {
+      static const double sphere_extrusion = 0.0625f;
+      Vector dir = (p2 - p1);
+      glyphs.addComet(p1-dir, p1, radius, resolution, node_color, node_color, sphere_extrusion);
       break;
+    }
     case RenderState::GlyphType::CONE_GLYPH:
-      glyphs.addCone(p1, p2, radius, resolution, true, node_color, node_color);
+      glyphs.addCone(p1, p2, radius, resolution, render_base1, node_color, node_color);
       break;
     case RenderState::GlyphType::ARROW_GLYPH:
-      glyphs.addArrow(p1, p2, radius, ratio, resolution, node_color, node_color);
+      glyphs.addArrow(p1, p2, radius, ratio, resolution, node_color, node_color, render_base1, render_base2);
       break;
     case RenderState::GlyphType::DISK_GLYPH:
-      glyphs.addDisk(p1, p2, radius, resolution, node_color, node_color);
+    {
+      Vector dir = (p2 - p1);
+      Point new_p2 = p1 + dir * radius * 2.0;
```

Comment:
This line isn't quite right.  The scaling of the length isn't staying consistent with the radius

Reply:
Scaling on this and rings should be good now